micov/_io.py

```diff
@@ -265,13 +265,19 @@ def _reader(sam):
     elif isinstance(sam, io.BytesIO):
         yield sam
     else:
-        with lzma.open(sam) as fp:
-            data = fp.read()
-            yield data
+        if sam.endswith('.sam.xz'):
```

Comment:
Thanks! Can these still be performed within context managers to ensure the files close? Eg:

```
if sam.endswith('.sam.xz'):
    with lzma.open(sam, mode='rt', encoding='utf-8') as fp:
        yield fp
```
?